### PR TITLE
Differentiate AggQC Validator Set from HighQC Validator Set

### PR DESCRIPTION
**Pull request by tholonious**

### CODE REVIEW

**Comment by tholonious** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/deso-protocol/core/917?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `feature/pos-syncing-and-steady-state`
  * **PR #916** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/916?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #917** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/917?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/deso-protocol/core/917?utm_source=stack-comment).

File: lib/pos_blockchain.go [hunk 1]
```diff
@@ -503,7 +503,7 @@ func (bc *Blockchain) processOrphanBlockPoS(block *MsgDeSoBlock) error {
 		}
 	}
 	// Okay now we have the validator set ordered by stake, we can validate the QC.
-	if err = bc.isValidPoSQuorumCertificate(block, validatorsByStake); err != nil {
+	if err = bc.isValidPoSQuorumCertificate(block, validatorsByStake, validatorsByStake); err != nil {
```
Comment (lazynina) [inline lib/pos_blockchain.go:506]: Does the second `validatorsByStake` need to change to represent the highQC validator set?

File: lib/pos_blockchain.go [hunk 2]
```diff
@@ -580,7 +580,7 @@ func (bc *Blockchain) validateLeaderAndQC(block *MsgDeSoBlock) (_passedSpamPreve
 	}
 
 	// Validate the block's QC. If it's invalid, we return true for failed spam prevention check.
-	if err = bc.isValidPoSQuorumCertificate(block, validatorsByStake); err != nil {
+	if err = bc.isValidPoSQuorumCertificate(block, validatorsByStake, validatorsByStake); err != nil {
```
Comment (lazynina) [inline lib/pos_blockchain.go:583]: same as above

File: lib/pos_blockchain.go (hunk 1 above)
Comment (lazynina) [reply lib/pos_blockchain.go:506]: Will be addressed in separate PR!

**Review by lazynina — APPROVED** [review]

File: lib/pos_blockchain.go (hunk 1 above)
Comment (tholonious) [reply lib/pos_blockchain.go:506]: Yes! I'll follow up on it in another PR. Every time we validate an aggregate QC, we need two different validator sets:
- The validator set at the incoming block's height
- The validator set at the incoming block's parent's height

**Pull request merged**